dashboard/views.py

```diff
@@ -1698,9 +1698,68 @@ def marshal(request):
     return render(request, 'marshal.html', context)
 
 
+def marshal_mab_edit_candidate(request):
+    initial = {'candidate': request.session.pop('candidate', None)}
+    form = MABEditCandidateForm(request.POST or None, initial=initial)
+
+    if request.method == 'POST':
+        if form.is_valid():
+            request.session['candidate'] = form.cleaned_data['candidate'].pk
+            return HttpResponseRedirect(reverse('dashboard:marshal_mab_edit'))
+
+    context = {
+        'form': form,
+    }
+
+    return render(request, 'marshal-mab-edit-candidate.html', context)
+
+
+def marshal_mab_edit(request):
+    candidate = Brother.objects.get(pk=request.session.get('candidate', None))
+    check_all = request.session.pop('check_all', False)
+
+    mab_form_list = []
+
+    brothers = Brother.objects.filter(brother_status='1')
+
+    for counter, brother in enumerate(brothers):
+        form = MeetABrotherEditForm(request.POST or None, prefix=counter+1, brother=brother.pk, mab_exists=MeetABrother.objects.filter(brother=brother, candidate=candidate).exists())
+        if check_all:
+            form.fields['update'].initial = True
+        mab_form_list.append(form)
+
+    if request.method == 'POST':
+        if 'submit' in request.POST:
+            if forms_is_valid(mab_form_list):
+                for counter, form in enumerate(mab_form_list):
```

Comment:
I wonder if we should make some sort of abstracted generator for these form lists that seem to keep popping up in what we are doing